modules/snmp-dest/snmpdest.c

```diff
@@ -221,12 +220,51 @@ void snmpdest_dd_set_community(LogDriver *d, const gchar *community)
   self->community = g_strdup(community);
 }
 
-void snmpdest_dd_set_engine_id(LogDriver *d, const gchar *eid)
+static gboolean
+_has_hexadecimal_prefix(const gchar *s)
+{
+  return (s[0] == '0') && (s[1] == 'x');
+}
+
+static gboolean
```

Comment:
did you check the possibility to use g_ascii_isxdigit() instead of an own implementation?